docs/user-guide/how_to_use_mpi_plugin.md

```diff
@@ -0,0 +1,76 @@
+# MPI plugin User Guide
+
+## Introduction
+
```

Comment:
Perhaps we can add a short paragrah to introduce the background and what have been done in this plugin for users. This doc can serve as an referenece: https://github.com/volcano-sh/volcano/blob/master/docs/user-guide/how_to_use_ssh_plugin.md